src/mbgl/style/expression/interpolate.cpp

```diff
@@ -151,29 +220,41 @@ ParseResult parseInterpolate(const Convertible& value, ParsingContext& ctx) {
     }
     
     assert(outputType);
-    
-    return outputType->match(
+
+    return createInterpolate(*outputType,
+                             *interpolator,
+                             std::move(*input),
+                             std::move(stops),
+                             ctx);
+}
+
+ParseResult createInterpolate(type::Type type,
+                              Interpolator interpolator,
+                              std::unique_ptr<Expression> input,
+                              std::map<double, std::unique_ptr<Expression>> stops,
+                              ParsingContext& ctx) {
```

Comment:
Should we move the check for stop inputs being in ascending order from https://github.com/mapbox/mapbox-gl-native/pull/12253/commits/644ed601794bf61dbf51c69718507552fc16c169#diff-f1eb8ec38d51de5044446f3a8d83f21aR204 to here?

Reply:
The stops are in a `std::map` at this point, so there's no way to check if they were ordered in the original input.